exp/controllers/awsmachinepool_controller.go

```diff
@@ -227,13 +228,58 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 	ec2Svc := r.getEC2Service(ec2Scope)
 	asgsvc := r.getASGService(clusterScope)
 
+	// Find existing ASG
+	asg, err := r.findASG(machinePoolScope, asgsvc)
+	if err != nil {
+		conditions.MarkUnknown(machinePoolScope.AWSMachinePool, expinfrav1.ASGReadyCondition, expinfrav1.ASGNotFoundReason, err.Error())
+		return ctrl.Result{}, err
+	}
+	if asg == nil {
+		machinePoolScope.Debug("asg not created yet")
+		r.Recorder.Eventf(machinePoolScope.AWSMachinePool, corev1.EventTypeNormal, expinfrav1.ASGNotFoundReason, "Unable to find matching ASG")
+	}
+
 	canUpdateLaunchTemplate := func() (bool, error) {
 		// If there is a change: before changing the template, check if there exist an ongoing instance refresh,
 		// because only 1 instance refresh can be "InProgress". If template is updated when refresh cannot be started,
 		// that change will not trigger a refresh. Do not start an instance refresh if only userdata changed.
 		return asgsvc.CanStartASGInstanceRefresh(machinePoolScope)
 	}
+	if asg == nil {
+		canUpdateLaunchTemplate = func() (bool, error) {
+			return true, nil
+		}
+	}
+
 	runPostLaunchTemplateUpdateOperation := func() error {
+		launchTemplateID := machinePoolScope.GetLaunchTemplateIDStatus()
```

Comment:
Shouldn't all this happen after the skipping instance refresh check? If instance refresh is completely disabled, none of these things should happen shouldn't they? Like, updating the tags.

Reply:
Ah, never mind. Of course, the ASG is using the LaunchTemplate. So we assure that it's up to date. Right?

Reply:
We must reconcile tags after updating the launch template, which is why this logic was moved into the beginning of `runPostLaunchTemplateUpdateOperation` - because we need to update our record of the tags on the launch template regardless of whether or not we are doing a refresh. The tags are a property of launch templates, not asgs.

I didn't write this logic, only relocated it from elsewhere in the codepath. Previously we only reconciled tags if the asg could be created, which is too late, because an error could have occurred.